plugins/lua/buildingplan/planneroverlay.lua

```diff
@@ -770,15 +857,125 @@ function PlannerOverlay:init()
         },
     }
 
+    local function make_is_selected_filter(idx)
+        return function () return self.selected_favorite == idx end
+    end
+
+    local favorites_panel = widgets.Panel{
+        view_id='favorites',
+        frame={t=15, l=0, r=0, h=9},
+        frame_style=gui.FRAME_INTERIOR_MEDIUM,
+        frame_background=gui.CLEAR_PEN,
+        visible=self:callback('show_favorites'),
+        subviews={
+            QuickFilter{idx=1, frame={t=0,l=0},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(1) },
+            QuickFilter{idx=2, frame={t=1,l=0},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(2) },
+            QuickFilter{idx=3, frame={t=2,l=0},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(3) },
+            QuickFilter{idx=4, frame={t=3,l=0},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(4) },
+            QuickFilter{idx=5, frame={t=4,l=0},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(5) },
+            QuickFilter{idx=6, frame={t=0,l=27},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(6) },
+            QuickFilter{idx=7, frame={t=1,l=27},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(7) },
+            QuickFilter{idx=8, frame={t=2,l=27},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(8) },
+            QuickFilter{idx=9, frame={t=3,l=27},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(9) },
+            QuickFilter{idx=0, frame={t=4,l=27},
+                        on_click_fn=self:callback("save_restore_filter"),
+                        is_selected_fn=make_is_selected_filter(0) },
+            widgets.HotkeyLabel{
+                frame={b=0, l=2},
+                label="foo",
+                key='CUSTOM_X',
+                on_activate=function() self.selected_favorite = (self.selected_favorite+1) % 10 end,
+            },
+            widgets.HotkeyLabel{
+                frame={b=0, l=3},
+                label="next/previous filter",
+                auto_width=true,
+                key='CUSTOM_SHIFT_X',
+                on_activate=function() self.selected_favorite = (self.selected_favorite-1) % 10 end,
+            },
```

Comment:
this logic is already available with a CycleHotkeyLabel with options with numeric values (can use blank labels): see https://github.com/DFHack/dfhack/blob/develop/plugins/lua/buildingplan/planneroverlay.lua#L592-L612